pkg/business/controller/imagenamespace/imagenamespace_health.go

```diff
@@ -0,0 +1,127 @@
+	s.mu.Lock()
+	defer s.mu.Unlock()
+	s.imageNamespaces.Insert(key)
+}
+
+func (c *Controller) startNamespaceHealthCheck(key string) {
+	if !c.health.Exist(key) {
+		c.health.Set(key)
+		go func() {
+			if err := wait.PollImmediateUntil(1*time.Minute, c.watchNamespaceHealth(key), c.stopCh); err != nil {
+				log.Error("Failed to wait poll immediate until", log.Err(err))
+			}
+		}()
+		log.Info("ImageNamespace phase start new health check", log.String("imageNamespace", key))
+	} else {
+		log.Info("ImageNamespace phase health check exit", log.String("imageNamespace", key))
+	}
+}
+
+// for PollImmediateUntil, when return true ,an err while exit
+func (c *Controller) watchNamespaceHealth(key string) func() (bool, error) {
+	return func() (bool, error) {
+		log.Debug("Check imageNamespace health", log.String("key", key))
+
+		if !c.health.Exist(key) {
+			return true, nil
+		}
+
+		projectName, imageNamespaceName, err := cache.SplitMetaNamespaceKey(key)
+		if err != nil {
+			log.Error("Failed to split meta imagenamespace key", log.String("key", key))
+			c.health.Del(key)
+			return true, nil
+		}
+
+		imageNamespace, err := c.client.BusinessV1().ImageNamespaces(projectName).Get(imageNamespaceName, metav1.GetOptions{})
+		if err != nil && errors.IsNotFound(err) {
+			log.Error("ImageNamespace not found, to exit the health check loop",
+				log.String("projectName", projectName), log.String("imageNamespaceName", imageNamespaceName))
+			c.health.Del(imageNamespaceName)
+			return true, nil
+		}
+		if err != nil {
+			log.Error("Check imageNamespace health, imageNamespace get failed",
+				log.String("projectName", projectName), log.String("imageNamespaceName", imageNamespaceName), log.Err(err))
+			return false, nil
+		}
+		// if status is terminated,to exit the  health check loop
+		if imageNamespace.Status.Phase == v1.ImageNamespaceTerminating || imageNamespace.Status.Phase == v1.ImageNamespacePending {
+			log.Warn("ImageNamespace status is terminated, to exit the health check loop",
+				log.String("projectName", projectName), log.String("imageNamespaceName", imageNamespaceName))
+			c.health.Del(imageNamespaceName)
+			return true, nil
+		}
+
+		if err := c.checkNamespaceHealth(imageNamespace); err != nil {
+			log.Error("Failed to check imageNamespace health",
+				log.String("projectName", projectName), log.String("imageNamespaceName", imageNamespaceName), log.Err(err))
+		}
+		return false, nil
+	}
+}
+
+func (c *Controller) checkNamespaceHealth(imageNamespace *v1.ImageNamespace) error {
+	namespaceList, err := c.registryClient.Namespaces().List(metav1.ListOptions{
+		FieldSelector: fmt.Sprintf("spec.tenantID=%s,spec.name=%s", imageNamespace.Spec.TenantID, imageNamespace.Spec.Name),
+	})
+	if err != nil {
+		return err
+	}
+	if len(namespaceList.Items) == 0 {
+		return fmt.Errorf("imagenamespace %s in tenant %s not exist", imageNamespace.Spec.Name, imageNamespace.Spec.TenantID)
+	}
+	return nil
```

Comment:
Here you need to check whether the image namespace is locked in the registry api